### PR TITLE
Add TEST_TMPDIR for couple bazel jobs

### PR DESCRIPTION
**Pull request by krzyzacy**
/assign @ixdy @spxtr

### CODE REVIEW

**Comment by spxtr** [comment]
Why not set this in the scenario? Right now it's set in the shell script for the jobs that still use those.

**Comment by ixdy** [comment]
setting `TEST_TMPDIR` in the scenario makes it harder to test locally, since it might not be what you want.

since we specify the mount point in the prow config, it makes sense to set the environment variable there too, IMO.

**Comment by ixdy** [comment]
I think @fejta also prefers keeping env magic out of scenarios...

**Comment by krzyzacy** [comment]
changed my mind too fast :smiling_imp: 

Originally was only want to fix the testgrid config upload job which does not use the scenario yet.

**Comment by spxtr** [comment]
OK, @ixdy convinced me.

File: prow/config.yaml [hunk 1]
```diff
@@ -80,6 +80,9 @@ presubmits:
         - "--service-account=/etc/service-account/service-account.json"
         - "--upload=gs://kubernetes-jenkins/pr-logs"
         # Bazel needs privileged mode in order to sandbox builds.
```
Comment (spxtr) [inline prow/config.yaml]: This comment is supposed to be above the `securityContext`.

Comment (krzyzacy) [inline prow/config.yaml]: fixed

**Pull request merged**